### ISSUE TITLE
Local Ton boot time

### ISSUE THREAD

**Issue by dkatzan** [neodix42/MyLocalTon#33]
Hi, looking at the logs while starting up a node, it looks like it take around a minute or more to finish loading
specifically I see:
`[MyLocalTon - genesis Status Monitor] INFO  org.ton.actions.MyLocalTon - genesis is not ready`
in intervals of 15 sec like 4-6 times before it's ready

is there anything I can do in the config\setup to get it up faster? 

context is that we are planning to use this for some tests as part of our CI pipeline, and it will add non negligible time to it

thx

**Comment by undron** [comment]
@dkatzan Hi! Could you run `MyLocalTon` in docker for a long time? I am trying to do the same thing right now and am having problems with block generation. After about 500 blocks all generation stops. It looks like `lite-client` could not get any data.

**Comment by dkatzan** [comment]
Still working on getting my setup going, but I did let it run for a while and recached a few thousands 
I'm running `java -jar MyLocalTon-arm64.jar nogui ton-http-api`

**Comment by undron** [comment]
Can you try to wait at least 600 blocks? Will it still work?

**Comment by dkatzan** [comment]
I did, I wrote it already reached a few thousands

**Comment by undron** [comment]
Could you please share your Dockerfile?

**Comment by undron** [comment]
![image](https://github.com/user-attachments/assets/1b2198bc-7e36-46d2-b1ce-c4ba6665487b)
So now it has passed 1139 blocks and is dead.

**Comment by undron** [comment]
Nevermind, I have the same problem on the local machine, despite of Docker

**Comment by dkatzan** [comment]
Sure, but I'm seeing some issues now as well, my genesis block got out of sync at block 483
![Screenshot 2024-08-06 at 17 21 54](https://github.com/user-attachments/assets/2596b7d3-ecff-4a4b-aa2b-dabcadac960e)

```
FROM ubuntu:24.04

WORKDIR /app

USER root

COPY $SERVICE_DIR .

RUN apt-get update
RUN apt-get install -y openjdk-17-jdk python3 python3-pip git curl vim cmake wget libsecp256k1-dev libsodium-dev
RUN pip3 install ton-http-api --break-system-packages

# MyLocalTon expects to find ton-http-api in a specific location.
RUN mkdir /root/.local
RUN mkdir /root/.local/bin
RUN cp /usr/local/bin/ton-http-api /root/.local/bin/

RUN wget https://github.com/neodix42/MyLocalTon/releases/latest/download/MyLocalTon-arm64.jar
RUN wget https://github.com/neodix42/MyLocalTon/releases/latest/download/MyLocalTon-x86-64.jar

RUN chmod +x /app/dockerentrypoint.sh

ENTRYPOINT ["/app/dockerentrypoint.sh"]
```



the `dockerentrypoint.sh` file contains

```
#!/bin/bash

java -jar MyLocalTon-arm64.jar nogui ton-http-api

wait
```

**Comment by dkatzan** [comment]
Which version of localton are u using?

**Comment by undron** [comment]
The last one, v111, but on v110 there was the same result.
Anyway, on Windows it pass more than 1200 blocks without any issues.

**Comment by dkatzan** [comment]
Are u sure u are having the same issue in v110?
I just tried twice with v111, and got out of sync after ±400 blocks
with v110 I'm already at 3000

**Comment by undron** [comment]
Not sure, will check tomorrow

**Comment by undron** [comment]
v110 failed after 1238 blocks.
![image](https://github.com/user-attachments/assets/bd6f15b8-bfa4-4190-bf04-decde1499df4)

**Comment by neodix42** [comment]
Definitely, the genesis node should not stop without any reason, I will take a look at this.
Regarding MyLocalTon in Docker - I see many people doing this, but I must say this is an inefficient way. 
I will come up with an optimal Docker version of MyLocalTon soon.

**Comment by dkatzan** [comment]
Thx, will try that and report back
